File: HRMS/Models/EmployeeUpdate.cs
using HRMS.Core.Models.Enums;

namespace HRMS.Models;

public class EmployeeUpdate
{
    public string FirstName { get; set; } = string.Empty;

    public string LastName { get; set; } = string.Empty;

    public string MaidenName { get; set; } = string.Empty;

    public string ParentName { get; set; } = string.Empty;

    public Gender Gender { get; set; }

    public string RegistrationNumber { get; set; } = string.Empty;

    public string PersonalIdentificationNumber { get; set; } = string.Empty;

    public string WorkerCode { get; set; } = string.Empty;

    public DateTime BirthDate { get; set; }

    public int? BirthPlaceId { get; set; }

    public string Address { get; set; } = string.Empty;

    public int? CityId { get; set; }

    public int? CitizenshipId { get; set; }

    public string Image { get; set; } = string.Empty;

    public string Email { get; set; } = string.Empty;

    public string Phone { get; set; } = string.Empty;

    public string Mobile { get; set; } = string.Empty;

    public string OfficePhone { get; set; } = string.Empty;

    public string Profession { get; set; } = string.Empty;

    public int? EducationId { get; set; }

    public int PreviousLOSYears { get; set; }

    public int PreviousLOSMonths { get; set; }

    public string BankAccount { get; set; } = string.Empty;

    public string Note { get; set; } = string.Empty;
}

File: HRMS.Desktop/lib/models/requests/employee_update.dart
import 'package:json_annotation/json_annotation.dart';

import '../enums/gender.dart';

part 'employee_update.g.dart';

@JsonSerializable()
class EmployeeUpdate {
  String firstName;
  String lastName;
  String maidenName;
  String parentName;
  Gender gender;
  String registrationNumber;
  String personalIdentificationNumber;
  String workerCode;
  DateTime birthDate;
  int? birthPlaceId;
  String address;
  int? cityId;
  int? citizenshipId;
  String image;
  String email;
  String phone;
  String mobile;
  String officePhone;
  String profession;
  int? educationId;
  int previousLOSYears;
  int previousLOSMonths;
  String bankAccount;
  String note;

  EmployeeUpdate(
    this.firstName,
    this.lastName,
    this.maidenName,
    this.parentName,
    this.gender,
    this.registrationNumber,
    this.personalIdentificationNumber,
    this.workerCode,
    this.birthDate,
    this.birthPlaceId,
    this.address,
    this.cityId,
    this.citizenshipId,
    this.image,
    this.email,
    this.phone,
    this.mobile,
    this.officePhone,
    this.profession,
    this.educationId,
    this.previousLOSYears,
    this.previousLOSMonths,
    this.bankAccount,
    this.note,
  );

  factory EmployeeUpdate.fromJson(Map<String, dynamic> json) =>
      _$EmployeeUpdateFromJson(json);

  Map<String, dynamic> toJson() => _$EmployeeUpdateToJson(this);
}